### DISCUSSION TITLE
dm+d coding system unavailable

### DISCUSSION THREAD

**Discussion by venexia** [opensafely/documentation#890] [Codelists] [answered]
Several of us ( @venexia, @kurttaylor, @RochelleKnight ) do not have the option to create dm+d codelists on OpenCodelists. Please can you advise?

![screenshot](https://user-images.githubusercontent.com/15819329/181501730-2eff82f3-5505-4e0e-8973-9f11bd35914d.png)

**Comment by brianmackenna** [comment, marked as answer]
Brief answer: It is not possible to build dm+d codelists natively on OpenCodelists - you should create a pseudoBNF codelists and then convert it to dm+d.

Long answer: The [NHS dictionary of medicines and devices (dm+d)](https://www.bennett.ox.ac.uk/blog/2019/08/what-is-the-dm-d-the-nhs-dictionary-of-medicines-and-devices/) is the terminology used to write prescriptions in England and must be used to query `with_these_medications` .dm+d does not have any human meaningful hierarchy or code structure so it is near impossible to make codelists directly from it as stands. However  [pseudoBNF codes](https://www.bennett.ox.ac.uk/blog/2017/04/prescribing-data-bnf-codes/#:~:text=The%20BNF%20codes%20from%20this,BNF%20a%20drug%20is%20from.) has a very human and clinically meaningful structure that allows development of codelists. PseudoBNF codes  are used for reimbursement of prescriptions and consequently they are already mapped to dm+d. I have been working on a blog to explain differences between [medicines terminologies here](https://www.bennett.ox.ac.uk/blog/2022/11/difference-between-bnf-dm-d-and-snomed-ct-codes/) - keen to have feedback on it.

In OpenSAFELY you should therefore make a pseudoBNF codelist and convert it to dm+d (see screenshot). Currently the workflow means you have to download it and reupload seperately as a dm+d list (which you'll need permissions for). We have an improvement awaiting prioritisation in [this issue](https://github.com/opensafely-core/opencodelists/issues/1334)

![image](https://user-images.githubusercontent.com/40460354/181741140-009f86d8-3f58-4aad-8b5d-5076fa995a49.png)

**Reply by venexia** [reply]
For completeness, please can you confirm the system that should be specified for `codelist_from_csv()` when using the above approach?

**Reply by alexwalkerepi** [reply]
The system should be `snomed`. Sounds odd, but: `dm+d` (which is what the `pseudoBNF` codelist is converted to) is an extension of `snomed`.

**Reply by venexia** [reply]
Thank you, Alex. That is what we thought but, given the confusion above, wanted to confirm!

**Comment by rc16** [comment]
Hi, I would like to upload a dm+d codelist but don't have the option to do this, as above. Can I be given the option to upload or is there someone I need to ask to do this?

**Reply by brianmackenna** [reply]
As an approved user you can ask `tech-support` in the OpenSAFELY slack channels
